src/app/components/utility/modal/modal.component.ts

```diff
@@ -230,33 +235,75 @@ export class ModalComponent implements OnInit {
+       this.isDisabled = e.target.value === this.user.first_name;
+     } else if (e.target.name === 'update_last_name') {
+       this.isDisabled = e.target.value === this.user.last_name;
+     } else if (e.target.name === 'update_affiliation') {
+       this.isDisabled = e.target.value === this.user.affiliation;
+     } else if (e.target.name === 'update_google_scholar_url') {
+       if (this.validURL(e.target.value)) {
+         this.isDisabled = e.target.value === this.user.google_scholar_url;
+
+         this.invalidFields = this.invalidFields.filter(element => element !== 'Google Scholar');
+         this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? `${this.invalidFields.join(' and ')} are` : `${this.invalidFields.join('')} is`} not valid` : null;
+       } else {
+         this.isDisabled = false; // In case they want to empty the field.
+
+         if (!(this.invalidFields.find(element => element === 'Google Scholar'))) { this.invalidFields.push('Google Scholar'); }
+         this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? `${this.invalidFields.join(' and ')} are` : `${this.invalidFields.join('')} is`} not valid` : null;
+       }
+     } else if (e.target.name === 'update_github_url') {
+      if (this.validURL(e.target.value)) {
+        this.isDisabled = e.target.value === this.user.github_url;
+
+        this.invalidFields = this.invalidFields.filter(element => element !== 'GitHub');
+        this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? `${this.invalidFields.join(' and ')} are` : `${this.invalidFields.join('')} is`} not valid` : null;
+       } else {
+        this.isDisabled = false; // In case they want to empty the field.
+
+        if (!(this.invalidFields.find(element => element === 'GitHub'))) { this.invalidFields.push('GitHub'); }
+        this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? `${this.invalidFields.join(' and ')} are` : `${this.invalidFields.join('')} is`} not valid` : null;
+       }
+     } else if (e.target.name === 'update_linkedin_url') {
+      if (this.validURL(e.target.value)) {
+        this.isDisabled = e.target.value === this.user.linkedin_url;
+
+        this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? `${this.invalidFields.join(' and ')} are` : `${this.invalidFields.join('')} is`} not valid` : null;
+        this.invalidFields = this.invalidFields.filter(element => element !== 'LinkedIn');
+       } else {
+        this.isDisabled = false; // In case they want to empty the field.
+
+        if (!(this.invalidFields.find(element => element === 'LinkedIn'))) { this.invalidFields.push('LinkedIn'); }
+        this.invalidFieldsAsText = this.invalidFields.length ? `${this.invalidFields.length > 1 ? `${this.invalidFields.join(' and ')} are` : `${this.invalidFields.join('')} is`} not valid` : null;
```

Comment:
This `and` for three field doesn't look good, can we add `,` in between if it's length is 3?